torchtext/experimental/vectors.py

```diff
@@ -1,7 +1,113 @@
 import csv
+import logging
+import os
+
 import torch
 from torch import Tensor
 import torch.nn as nn
+from tqdm import tqdm
+
+from torchtext.utils import (
+    download_from_url,
+    extract_archive
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _load_token_and_vectors_from_file(file_path):
+    stoi, tokens, vectors, dup_tokens = {}, [], [], []
+    with open(file_path, 'rb') as f:
+        for line in tqdm(f):
+            token, entries = line.rstrip().split(b" ", 1)
+            vector = torch.tensor([float(c) for c in entries.split(b" ")], dtype=torch.float)
+
+            try:
+                if isinstance(token, bytes):
+                    token = token.decode('utf-8')
+            except UnicodeDecodeError:
+                logger.info("Skipping non-UTF8 token {}".format(repr(token)))
+                continue
+
+            if token in stoi:
```

Comment:
This seems like something that should be done outside of this function as a separate operation. Also, doesn't the Vectors class already do this? If we mean to remove duplicate elements because we expect it to be for a particular set of Vectors, we should only remove duplicates in that case, because in other cases it might be a bug.

Even more so, ideally we know what element we expect to be duplicated and explicitly check plus make a choice about this.